api/websocket/server/server.go

```diff
@@ -192,6 +193,24 @@ func (ws *WsServer) registryMethod() {
 			return api.RespPacking(api.NodeInfo(wsAddr, rpcAddr, pubkey, id), errcode.WRONG_NODE)
 		}
 
+		// client auth
+		signature, ok := cmd["Signature"]
+		if ok { // if client send Signature, then check it
+			strSignature := signature.(string)
+			byteSignature, _ := hex.DecodeString(strSignature)
+
+			var challenge []byte
+			sess := cmd["session"].(*session.Session)
```

Comment:
Have you tried `ws.SessionList.GetSessionsById`? Then you can remove `cmd["session"]`